### DISCUSSION TITLE
Multiple unknown packages

### DISCUSSION THREAD

**Discussion by TikoyX** [Homebrew/discussions#3716] [Getting started] [answered]
Hello everyone

I am running a mac with OS version 12.6, Homebrew version 3.6.3. 

I currently have about 70 packages installed, but it is only about 10 of the packages i can recognise as something i have installed.
There was already preinstalled packages in there when i installed Homebrew (as far as i remember) and the list of unknown packages just increses over time.

I have tried the `brew autoremove` which did remove some of it, but is there a way i can see what packages are linked together? It is starting to get a bit weird and i dont know if i should just start to delete packages i don't know.

Thank you in advance!

**Comment by SMillerDev** [comment, marked as answer]
You can run `brew deps --tree <name>` for the packages you installed to see what each package pulled in as a dependency.

**Reply by TikoyX** [reply]
aaaarh awesome! Thank you very much! 
Wish there was a command for just showing "main packages" and not all the stuff that was "side installed"

**Reply by TikoyX** [reply]
Apparently that should be `brew leaves` but that still gives me packages i don't recognise

**Reply by unikitty37** [reply]
Is there a command where I can get Homebrew to tell me which package installed a particular file? `brew deps` doesn't seem to support this.

I'm hoping to type something like `brew what-installed /opt/homebrew/bin/imginfo` and get the package name output.

**Reply by SMillerDev** [reply]
Since it's all symlinks, `ls -al /opt/Homebrew/bin/<something>` will tell you what package the symlink points

**Reply by unikitty37** [reply]
🤦 

Thanks! I have `ls` aliased to `ls -CF`, as well, so I should have noticed the `@` at the end…
